File: server/Models/Recipe.cs
namespace allspice.Models;

public class Recipe
{ // use upper pascal case
  public int Id { get; set; }
  public DateTime CreatedAt { get; set; }
  public DateTime UpdatedAt { get; set; }
  public string Title { get; set; }
  public string Instructions { get; set; }
  public string Img { get; set; }
  public string Category { get; set; }
  public string CreatorId { get; set; }

  public Account Creator { get; set; }
}

public class FavoriteRecipe : Recipe
{
  // public Account Creator { get; set; } not needed since recipe inherits it
  // public Recipe Recipe { get; set; }


  public int FavoriteId { get; set; }
}

File: client/src/models/Recipe.js
import { DatabaseDefaults } from "./DatabaseDefaults.js";

export class Recipe extends DatabaseDefaults{
  constructor(d){ // data
    super(d)
    this.title = d.title
    this.instructions = d.instructions
    this.img = d.img
    this.category = d.category
    this.creatorId = d.creatorId
    this.creator = d.creator

  }
}